pkg/operator/endpoint_status_builder.go

```diff
@@ -68,24 +68,145 @@ func (b *scrapeEndpointBuilder) add(target *prometheusv1.TargetsResult) error {
 	return nil
 }
 
+type ScrapePoolKind uint8
+
+const (
+	ScrapePoolInvalid ScrapePoolKind = iota
+	ScrapePoolKubelet
+	ScrapePoolPodMonitoring
+	ScrapePoolClusterPodMonitoring
+	ScrapePoolNodeMonitoring
+)
+
+func parseScrapeKind(kind string) (ScrapePoolKind, error) {
```

Comment:
question: Why use this function at all (with the extra `iota` declarations)? What's wrong with rewriting the case statement to use the prefix directly?
i.e.
```go
kind := split[1]
switch kind {
case "kubelet":
	// lala
case "PodMonitoring":
	// lala
case "ClusterPodMonitoring":
	// lala
case "NodeMonitoring":
	// lala
default:
```

Reply:
Good point. Thanks!